OrangebeardLogger.cs

```diff
@@ -292,47 +292,85 @@ private void PopulateAttachmentData(ref string message, ref string attachmentMim
 
         private bool HandlePotentialStartFinishLog(IDictionary<string, string> info)
         {
```

Comment:
This method has become very long. If possible, we  should see if we can extract more of it to separate methods.

You did a good job extracting  code to `UpdateTree`, but I think we should try and extract a little more.